Work around the following compilation error observed with clang-18:

  ../../base/containers/flat_map.h:330:49: error: out-of-line definition of 'try_emplace' does not match any declaration in 'flat_map<Key, Mapped, Compare, Container>'
    330 | auto flat_map<Key, Mapped, Compare, Container>::try_emplace(K&& key,
        |                                                 ^~~~~~~~~~~
  ../../base/containers/flat_map.h:344:49: error: out-of-line definition of 'try_emplace' does not match any declaration in 'flat_map<Key, Mapped, Compare, Container>'
    344 | auto flat_map<Key, Mapped, Compare, Container>::try_emplace(const_iterator hint,
        |                                                 ^~~~~~~~~~~

--- a/base/containers/flat_map.h
+++ b/base/containers/flat_map.h
@@ -234,11 +234,9 @@ class flat_map : public ::base::internal
   iterator insert_or_assign(const_iterator hint, K&& key, M&& obj);
 
   template <class K, class... Args>
-    requires(std::is_constructible_v<key_type, K &&>)
   std::pair<iterator, bool> try_emplace(K&& key, Args&&... args);
 
   template <class K, class... Args>
-    requires(std::is_constructible_v<key_type, K &&>)
   iterator try_emplace(const_iterator hint, K&& key, Args&&... args);
 
   // --------------------------------------------------------------------------
@@ -324,9 +322,6 @@ auto flat_map<Key, Mapped, Compare, Cont
 
 template <class Key, class Mapped, class Compare, class Container>
 template <class K, class... Args>
-  requires(std::is_constructible_v<
-           typename flat_map<Key, Mapped, Compare, Container>::key_type,
-           K &&>)
 auto flat_map<Key, Mapped, Compare, Container>::try_emplace(K&& key,
                                                             Args&&... args)
     -> std::pair<iterator, bool> {
@@ -338,9 +333,6 @@ auto flat_map<Key, Mapped, Compare, Cont
 
 template <class Key, class Mapped, class Compare, class Container>
 template <class K, class... Args>
-  requires(std::is_constructible_v<
-           typename flat_map<Key, Mapped, Compare, Container>::key_type,
-           K &&>)
 auto flat_map<Key, Mapped, Compare, Container>::try_emplace(const_iterator hint,
                                                             K&& key,
                                                             Args&&... args)
